chia/wallet/derive_keys.py

```diff
@@ -110,3 +112,26 @@ def find_authentication_sk(all_sks: List[PrivateKey], owner_pk: G1Element) -> Op
                 # NOTE: ONLY use 0 for authentication key index to ensure compatibility
                 return master_sk_to_pooling_authentication_sk(sk, uint32(pool_wallet_index), uint32(0))
     return None
+
+
+def match_address_to_sk(
+    sk: PrivateKey, addresses_to_search: List[bytes32], max_ph_to_search: int = 500
+) -> Dict[bytes32, bool]:
+    """Checks the list of given address is a derivation of the given sk within the given number of derivations"""
+    if sk is None or not addresses_to_search:
+        return {}
+
+    found_dict: Dict[bytes32, bool] = {}
+    for i in range(max_ph_to_search):
+        phs = [
+            create_puzzlehash_for_pk(master_sk_to_wallet_sk(sk, uint32(i)).get_g1()),
+            create_puzzlehash_for_pk(master_sk_to_wallet_sk_unhardened(sk, uint32(i)).get_g1()),
+        ]
+        for address in addresses_to_search:
```

Comment:
I'd make a copy of addresses_to_search and remove entries as they're found.

Reply:
I thought about this, but at the moment, it's only used for a List of 2 and for small lists, I thought that overhead wouldn't be worth it. Although there is some problem here if the list contains any duplicated addresses, this will loop around more than required as the length check doesn't work in this case, so I'll need to fix that.